pkg/topolvm-scheduler/cmd/access_log.go

```diff
@@ -0,0 +1,60 @@
+package cmd
+
+import (
+	"context"
+	"net"
+	"net/http"
+	"time"
+
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+type accessLogResponseWriter struct {
+	http.ResponseWriter
+	statusCode int
+	size       int
+}
+
+func (w *accessLogResponseWriter) Write(data []byte) (int, error) {
+	n, err := w.ResponseWriter.Write(data)
+	w.size += n
+	return n, err
+}
+
+func (w *accessLogResponseWriter) WriteHeader(statusCode int) {
+	w.statusCode = statusCode
+	w.ResponseWriter.WriteHeader(statusCode)
+}
+
+func accessLogHandler(ctx context.Context, next http.Handler) http.Handler {
```

Comment:
Would you add some tests for accessLogHandler?

Reply:
I wrote `TestAccessLogHandler`, the test for `accessLogHandler`. It's similar to [cybozu-go/well's tests](https://github.com/cybozu-go/well/blob/8cbd22dd58421e97e3cd8a9ecc0ce65000dd682f/http_test.go#L221), but there are some differences:

- I omitted tests for `logged_at` because encoding of time is done automatically by zap.
- I omitted tests for `severity` because logr doesn't support warning level logging and `accessLogHandler` only uses info level.